flutter/lib/ui/painting.dart

```diff
@@ -280,20 +316,101 @@ class Paint {
   ///
   /// This color is not used when compositing. To colorize a layer, use
   /// [colorFilter].
-  Color color = _kDefaultPaintColor;
-  static const Color _kDefaultPaintColor = const Color(0xFF000000);
+  Color get color {
+    final int encoded = _data.getInt32(_kColorOffset, _kFakeHostEndian);
+    return new Color(encoded ^ _kColorDefault);
+  }
+  set color(Color value) {
+    assert(value != null);
```

Comment:
since this is the only one that doesn't accept null, I think we should make it accept null.